src/cirrus/lib/process_payload.py

```diff
@@ -198,7 +207,7 @@ def __call__(self, wfem) -> str | None:
         if not payload_bucket:
             raise ValueError("env var CIRRUS_PAYLOAD_BUCKET must be defined")
 
-        arn = os.getenv("CIRRUS_BASE_WORKFLOW_ARN") + self.process["workflow"]
+        state_machine_arn, _, execution_name = execution_arn.rpartition(":")
```

Comment:
A piece that came up in debugging:  the workflow name is documented in two locations.

1. `self.process["workflow"]`
2. `re.sub(r".*/workflow-(?P<workflow>[^/][^/]*)/.*", r"\g<workflow>", payload["id"])`

I in implementing `gen_execution_arn` I had fallen back to the latter, but this line shows that the former is authoritative.